core/charm/repository/charmhub.go

```diff
@@ -142,9 +138,16 @@ func (c *CharmHubRepository) ResolveWithPreferredChannel(charmURL *charm.URL, re
 	// Ensure we send the updated charmURL back, with all the correct segments.
 	revision := res.Entity.Revision
 	resCurl := charmURL.
-		WithSeries(chSuggestedOrigin.Platform.Series).
 		WithArchitecture(chSuggestedOrigin.Platform.Architecture).
 		WithRevision(revision)
+	// TODO(wallyworld) - does charm url still need a series?
```

Comment:
charm blobs are specific to a platform, it's why the charm url has the series today. once we move from a charm url to a charm identifier, (such as the download string), we can remove the series from the url - or get rid of url in general.

Reply:
we track origin internally and use origin to query charmhub right? and series is no longer sufficient on its own to specify the blob anyway, as we need arch too. so do we really need it? i mean ch:jammy/ubuntu is ambiguous since arch is missing?